src/main/java/com/tozny/sdk/RealmApi.java

```diff
@@ -247,10 +247,11 @@ private boolean dispatchUserExists(UserExistsRequest req) throws ToznyApiExcepti
     }
 
     private boolean dispatchUserPush(UserPushRequest request) throws ToznyApiException {
-        ToznyApiResponse<Boolean> response = protocol.<ToznyApiResponse<Boolean>>dispatch(
-                request, new TypeReference<ToznyApiResponse<Boolean>>() {});
-        String ret = response.getReturn();
-        return ret != null && ret.equals("true");
+        ToznyApiResponse<Boolean[]> response = protocol.<ToznyApiResponse<Boolean[]>>dispatch(
+                request, new TypeReference<ToznyApiResponse<Boolean[]>>() {});
+
+        Boolean[] results = response.getResult();
+        return results != null && results.length == 1 && results[0];
```

Comment:
This expression doesn't distinguish wrong results from the result of the operation.

* `results ==  null` - Server error
* `results.length != 1` - Server error
* `results[0] == null` - Server error (and missing)
* `results[0]` - Result of the operation  

Does the SDK distinguish server errors somehow? I think the right thing is to throw an exception, but I would rather keep this consistent with existing code (i.e., not have to modify all of it).